format/puffin-spec.md

```diff
@@ -123,6 +123,54 @@ The blob metadata for this blob may include following properties:
 
 - `ndv`: estimate of number of distinct values, derived from the sketch.
 
+#### `delete-vector-v1` blob type
+
+A serialized delete vector (bitmap) that represents the positions of rows in a
+file that are deleted.  A set bit at position P indicates that the row at
+position P is deleted.
+
+The vector supports positive 64-bit positions (the most significant bit must be
+0), but is optimized for cases where most positions fit in 32 bits by using a
+collection of 32-bit Roaring bitmaps.  64-bit positions are divided into a
+32-bit "key" using the most significant 4 bytes and a 32-bit sub-position using
+the least significant 4 bytes. For each key in the set of positions, a 32-bit
+Roaring bitmap is maintained to store a set of 32-bit sub-positions for that
+key.
+
+To test whether a certain position is set, its most significant 4 bytes (the
+key) are used to find a 32-bit bitmap and the least significant 4 bytes (the
+sub-position) are tested for inclusion in the bitmap. If a bitmap is not found
+for the key, then it is not set.
+
+The serialized blob contains:
+* The length of the vector and magic bytes stored as 4 bytes, big-endian
```

Comment:
What is length here?  Is it potentially already captured by metadata (e.g. offset/length)

Reply:
It is the length of the bitmap serialized using the portable serialization + 4 extra bytes for the magic number.

Reply:
This seems fairly redundant with offset and length information that we are already storing, even for compatibility purposes I'm not sure why it is needed?

Reply:
My issue (as noted on the other thread) isn't that the redundancy bothers me, it's that we are probably going to have two values for length. One that includes the CRC in the manifest, and one that doesn't within the blob itself. That's a bit odd.

Reply:
It is true. Puffin footer and manifests will capture the length of the entire blob (4 bytes for bitmapData length, magic number, bitmap, checksum) and this length will only cover the magic number and bitmap.

Reply:
Suggest to use 'combined length'.  On first read , it could be mis-interpreted as 'length :: magic bytes'

Reply:
Updated to use Szehon's suggestion.